tests/test_Custom.cpp

```diff
@@ -0,0 +1,78 @@
+#include <gtsam/geometry/Cal3_S2Stereo.h>
+#include <gtsam/geometry/StereoPoint2.h>
+#include <gtsam/slam/StereoFactor.h>
+#include <jrl/Dataset.h>
+#include <jrl/DatasetBuilder.h>
+#include <jrl/Parser.h>
+#include <jrl/Writer.h>
+#include <jrl/IOMeasurements.h>
+#include <jrl/IOValues.h>
+
+#include "gtest/gtest.h"
+
+using gtsam::symbol_shorthand::X;
+using gtsam::symbol_shorthand::V;
+using gtsam::symbol_shorthand::B;
+
+#define EXPECT_MATRICES_EQ(M_actual, M_expected) \
+  EXPECT_TRUE(M_actual.isApprox(M_expected, 1e-6)) << "  Actual:\n" << M_actual << "\nExpected:\n" << M_expected
+
+
+TEST(Custom, Factor){
+    // Make dummy factors
+    gtsam::noiseModel::Gaussian::shared_ptr cov = gtsam::noiseModel::Gaussian::Covariance(Eigen::Matrix3d::Identity());
+    gtsam::PriorFactor<gtsam::Pose2> prior(X(0), gtsam::Pose2::Identity(), cov);
+    std::string customType = "CustomFactor";
+
+    // Make dataset
+    gtsam::NonlinearFactorGraph graph;
+    graph.push_back(prior);
+    jrl::DatasetBuilder builder("test", {'a'});
+    builder.addEntry('a', 0, graph, {customType});
+
+    // Make custom writer/parser (Copied from prior information)
+    jrl::Writer writer;
+    writer.registerMeasurementSerializer(customType, [customType](gtsam::NonlinearFactor::shared_ptr& factor) { return jrl::io_measurements::serializePrior<gtsam::Pose2>(&jrl::io_values::serialize<gtsam::Pose2>, customType, factor); });
+
+    jrl::Parser parser;
+    parser.registerMeasurementParser(customType, [](const json& input){ return jrl::io_measurements::parsePrior<gtsam::Pose2>(&jrl::io_values::parse<gtsam::Pose2>, input); });
+
+    // Send it round trip!
+    writer.writeDataset(builder.build(), "custom.jrl");
```

Comment:
It gets written into the 'build/tests' directory. Currently does not get removed, but that'd be an easy fix.

I'll plan on exposing `serializeMeasurements` and `parseMeasurements` and test via those instead.

I would argue that it'd be worthwhile to have some tests write to file, otherwise the functions `writeDataset`, `writeJson`, etc won't ever be tested which is the interface most users will be using. That probably belongs in another test though!